run_test.sh

```diff
@@ -380,6 +381,50 @@ EOF
 sig_key${MODULE_SERIAL}
 EOF
     fi
+
+    # If the key can't be extracted from module, the hash algorithm is also unknown
+    if [[ "${SIG_KEY// /}" != "sig_key:" ]] && grep -q '^CONFIG_MODULE_SIG_HASH=' "/lib/modules/${KERNEL_VER}/build/.config"; then
+        echo 'Building the test module using a different hash algorithm'
+        if grep -q '^CONFIG_MODULE_SIG_HASH="sha512"' "/lib/modules/${KERNEL_VER}/build/.config"; then
+            ALTER_HASH="sha256"
+        else
+            ALTER_HASH="sha512"
+        fi
+        run_with_expected_output dkms uninstall -k "${KERNEL_VER}" -m dkms_test -v 1.0 << EOF
+Module dkms_test-1.0 for kernel ${KERNEL_VER} ($(uname -m)).
+Before uninstall, this module version was ACTIVE on this kernel.
+
+dkms_test.ko${mod_compression_ext}:
+ - Uninstallation
+   - Deleting from: /lib/modules/${KERNEL_VER}/${expected_dest_loc}/
+ - Original module
+   - No original module was found for this module on this kernel.
+   - Use the dkms install command to reinstall any previous module version.
+EOF
+        echo "CONFIG_MODULE_SIG_HASH=\"${ALTER_HASH}\"" > /tmp/dkms_test_kconfig
+        run_with_expected_output dkms build -k "${KERNEL_VER}" -m dkms_test -v 1.0 --config /tmp/dkms_test_kconfig --force << EOF
+
+Building module:
+Cleaning build area...
+make -j$(nproc) KERNELRELEASE=${KERNEL_VER} -C /lib/modules/${KERNEL_VER}/build M=/var/lib/dkms/dkms_test/1.0/build...
+${SIGNING_MESSAGE}Cleaning build area...
+EOF
+        run_with_expected_output dkms install -k "${KERNEL_VER}" -m dkms_test -v 1.0 << EOF
+
+dkms_test.ko${mod_compression_ext}:
+Running module version sanity check.
+ - Original module
+   - No original module exists within this kernel
+ - Installation
+   - Installing to /lib/modules/${KERNEL_VER}/${expected_dest_loc}/
+depmod...
+EOF
+        run_with_expected_output sh -c "modinfo '/lib/modules/${KERNEL_VER}/${expected_dest_loc}/dkms_test.ko${mod_compression_ext}' | grep '^sig_hashalgo:' | sed 's/sig_hashalgo: *//'" << EOF
```

Comment:
No need to filter out the "sig_hashalgo: " piece, just add it to the expectation below?

Reply:
OK